R/loadFromMothur.R

```diff
@@ -0,0 +1,206 @@
+#' Import Mothur results to \code{TreeSummarizedExperiment}
+#'
+#' @param featureTableFile a single \code{character} value defining the file
+#'   path of the feature table to be imported.
+#'
+#' @param taxonomyTableFile a single \code{character} value defining the file
+#'   path of the taxonomy table to be imported. (default:
+#'   \code{taxonomyTableFile = NULL}).
+#'
+#' @param sampleMetaFile a single \code{character} value defining the file path
+#'   of the sample metadata to be imported. (default: \code{sampleMetaFile = NULL}).
+#'
+#' @param featureNamesAsRefSeq \code{TRUE} or \code{FALSE}: Should the feature
+#'   names of the feature table be regarded as reference sequences? This setting
+#'   will be disregarded, if \code{refSeqFile} is not \code{NULL}. If the
+#'   feature names do not contain valid DNA characters only, the reference
+#'   sequences will not be set.
+#'
+#' @param refSeqFile a single \code{character} value defining the file path of
+#'   the reference sequences for each feature. (default: \code{refSeqFile =
+#'   NULL}).
+#'
+#' @param phyTreeFile a single \code{character} value defining the file path of
+#'   the phylogenetic tree. (default: \code{phyTreeFile = NULL}).
```

Comment:
I am not really familiar with mothur output. Are those data types available from a typical mothur run?

If not, please remove them for clarity.

Reply:
I'm not familiar either with Mothur, so that is why I basically tried to just implemented the microbiome's function

It seems that Mothur output files can include:

- assay: https://mothur.org/wiki/shared_file/
- rowData: https://mothur.org/wiki/taxonomy_file/ or https://mothur.org/wiki/constaxonomy_file/
- colData: https://mothur.org/wiki/design_file/

(Full file type list: https://mothur.org/wiki/tags/#file_types)

So, these file types above should be allowed as input?
I should probably try to find more presentative data so I don't have to do this blindfolded

Reply:
> So, these file types above should be allowed as input?

I guess so

Reply:
Yes these are standard outputs of mother.

Reply:
Having read the mothur man page, make sure to use their nomenclature. From the names you are using right now, it is for example not clear, in which file the count data is expected to be.